mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,384 @@
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/zBoldTagHelperCopy.cs
+   :language: c#
+   :lines: 5-6
+ 
+When multiple attributes are added to the same statement, the runtime treats them as a logical-AND. For example, in the code below, an HTML element must contain both a bold tag and a bold attribute to match.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/zBoldTagHelperCopy2.cs
+   :language: c#
+   :lines: 5
+
+For a good example of a bootstrap progress bar that targets a tag and an attribute, see `Creating custom MVC 6 Tag Helpers <http://www.davepaquette.com/archive/2015/06/22/creating-custom-mvc-6-tag-helpers.aspx>`__.
+
+You can also use the ``TargetElement`` to change the name of the targeted element. For example if you wanted the ``BoldTagHelper`` to target ``<MyBold>`` tags, you would use the following attribute:
+
+.. code-block:: c#
+
+	    [TargetElement("MyBold")]
+
+Web site information Tag Helper
+------------------------------------
+
+#. Add a *Models* folder.
+#. Add the following ``WebsiteContext`` class to the *Models* folder:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/Models/WebsiteContext.cs
+    :language: c#
+
+3. Add the following ``WebsiteInformationTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/WebsiteInformationTagHelper.cs
+    :language: c#
+
+Notes: 
+
+- As mentioned previously, the ASP.NET 5 runtime translates Pascal-cased C# class names and properties for tag helpers into `lower kebab case <http://c2.com/cgi/wiki?KebabCase>`__. Therefore, to use the ``WebsiteInformationTagHelper`` in Razor, you'll write ``<website-information />``.
+- We are not explicitly identifying the target element with the ``TargetElement`` attribute, so the default of ``website-information`` will be targeted. If you applied the following attribute (note it's not kebab case but matches the class name):
+
+.. code-block:: c#
+
+	    [TargetElement("WebsiteInformation")]
+
+The lower kebab case tag ``<website-information />`` would not match. If you want use the ``TargetElement`` attribute, you would use kebab case as shown below:
+
+.. code-block:: c#
+
+	    [TargetElement("Website-Information")]
+
+- `Void elements <http://www.w3.org/TR/html5/syntax.html#void-elements>`_ (that is, an element with a self-closing tag) have no content. For this example, the Razor markup will use a self-closing tag, but the tag helper will be creating a `section <http://www.w3.org/TR/html5/sections.html#the-section-element>`__ element (which is not self-closing and we are writing content inside the ``section`` element). Therefore, we need to set the ``TagMode`` to ``StartTagAndEndTag`` to write output. Alternatively, you can comment out the line setting ``TagMode`` and write markup with a closing tag. (Example markup is provided later in this tutorial.)
```

Comment:
[here] (https://github.com/aspnet/Docs/pull/419/files#r38722824) takes me to the start of PR419